### PR TITLE
Add environment variable to start

### PR DESCRIPTION
**Pull request by babsey**
This PR adds environment variable in `entrypoint.sh` which mode should be executed.

### CODE REVIEW

File: src/3.1/entrypoint.sh [hunk 1]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+        mkdir $1
+    fi
+    cd $1
+}
 
-if [[ "$1" = 'notebook' ]]; then
-    cd /opt/data
-    exec jupyter-notebook --ip="${IP_ADDRESS}" --port=8080 --no-browser --allow-root
-fi
+MODE="${mode:-$1}"
```
Comment (terhorstd) [inline src/3.1/entrypoint.sh:27]: Why did you choose a non-standard lower-case variable name over the one in our discussion? Wouldn't a more "scoped" name also help to reduce misunderstandings when many variables are being passed in?

File: src/latest/entrypoint.sh [hunk 2]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+        mkdir $1
+    fi
+    cd $1
+}
 
-if [[ "$1" = 'notebook' ]]; then
-    cd /opt/data
-    exec jupyter-notebook --ip="${IP_ADDRESS}" --port=8080 --no-browser --allow-root
-fi
+MODE="${mode:-$1}"
```
Comment (terhorstd) [inline src/latest/entrypoint.sh:27]: same here, of course

File: src/3.1/entrypoint.sh [hunk 3]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
-if [[ ! -d /opt/data ]]; then
-    mkdir /opt/data
-fi
+# Create folder if not existed and then change the directory.
+mkdir_cd () {
+    if [[ ! -d $1 ]]; then
+        mkdir $1
+    fi
+    cd $1
+}
```
Comment (terhorstd) [inline src/3.1/entrypoint.sh:25]: I'm not sure if it makes sense to factor this out. It is the slow version of `mkdir -p $1; cd $1;`, the `if` just makes it slow and the function adds complexity that the operation does not require. My apologies, I didn't see that during our discussion.
```suggestion
```

File: src/3.1/entrypoint.sh [hunk 4]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
-if [[ "$1" = 'notebook' ]]; then
-    cd /opt/data
-    exec jupyter-notebook --ip="${IP_ADDRESS}" --port=8080 --no-browser --allow-root
-fi
+MODE="${mode:-$1}"
+if [[ "${MODE}" = 'interactive' ]]; then
+    read -p "Your python script: " name
+    echo Starting: $name
+    # Start
+    mkdir_cd /opt/data
```
Comment (terhorstd) [inline src/3.1/entrypoint.sh:32]: Using `mkdir -p` has exactly the effect of the `if` in the function.
```suggestion
    mkdir -p /opt/data
    cd /opt/data
```

File: src/3.1/entrypoint.sh [hunk 5]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+    exec python3 /opt/data/$name
 
-if [[ "$1" = 'jupyterlab' ]]; then
-    cd /opt/data
+elif [[ "${MODE}" = 'jupyterlab' ]]; then
+    mkdir_ch /opt/data
```
[the remark this answers is not in the document]
Comment (terhorstd) [reply src/3.1/entrypoint.sh:36]: so actually also here
```suggestion
    mkdir -p /opt/data
    cd /opt/data
```

File: src/3.1/entrypoint.sh [hunk 6]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+    export NEST_SERVER_MODULES=nest,numpy
     exec nest-server start -o -h 0.0.0.0 -p 5000 -u 65534
-fi
 
-if [[ "$1" = 'nest-desktop' ]]; then
-    cd /opt/data
-    exec /root/.local/bin/nest-desktop start
-fi
+elif [[ "${MODE}" = 'notebook' ]]; then
+    mkdir_cd /opt/data
```
Comment (terhorstd) [inline src/3.1/entrypoint.sh:48]:
```suggestion
    mkdir -p /opt/data
    cd /opt/data
```

File: src/latest/entrypoint.sh [hunk 7]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
-if [[ ! -d /opt/data ]]; then
-    mkdir /opt/data
-fi
+# Create folder if not existed and change the director to /opt/data.
+mkdir_cd () {
+    if [[ ! -d $1 ]]; then
+        mkdir $1
+    fi
+    cd $1
+}
```
Comment (terhorstd) [inline src/latest/entrypoint.sh:25]:
```suggestion
```

File: src/latest/entrypoint.sh [hunk 8]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+    read -p "Your python script: " name
+    echo Starting: $name
+    # Start
+    mkdir_cd /opt/data
+    exec python3 /opt/data/$name
 
-if [[ "$1" = 'jupyterlab' ]]; then
-    cd /opt/data
+elif [[ "${MODE}" = 'jupyterlab' ]]; then
+    mkdir_ch /opt/data
```
Comment (terhorstd) [inline src/latest/entrypoint.sh:36]:
```suggestion
    mkdir -p /opt/data
    cd /opt/data
```

File: src/latest/entrypoint.sh [hunk 9]
```diff
@@ -15,40 +15,39 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+    export NEST_SERVER_MODULES=nest,numpy
     exec nest-server start -o -h 0.0.0.0 -p 5000 -u 65534
-fi
 
-if [[ "$1" = 'nest-desktop' ]]; then
-    cd /opt/data
-    exec /root/.local/bin/nest-desktop start
-fi
+elif [[ "${MODE}" = 'notebook' ]]; then
+    mkdir_cd /opt/data
```
Comment (terhorstd) [inline src/latest/entrypoint.sh:48]:
```suggestion
    mkdir -p /opt/data
    cd /opt/data
```

File: src/3.1/entrypoint.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (steffengraber) [reply src/3.1/entrypoint.sh:36]: @terhorstd I agree. The function brings no real advantage.

File: src/3.1/entrypoint.sh (hunk 1 above)
Comment (babsey) [reply src/3.1/entrypoint.sh]: Indeed, I changed it to `NEST_CONTAINER_MODE`.

File: src/3.1/entrypoint.sh [hunk 10]
```diff
@@ -15,40 +15,32 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+elif [[ "${MODE}" = 'nest-desktop' ]]; then
+    exec /root/.local/bin/nest-desktop start -h 0.0.0.0 -p 8000
 
-if [[ "$1" = 'nest-desktop' ]]; then
-    cd /opt/data
-    exec /root/.local/bin/nest-desktop start
-fi
+elif [[ "${MODE}" = 'nest-server' ]]; then
+    export NEST_SERVER_RESTRICTION_OFF=true
+    export NEST_SERVER_MODULES=nest,numpy
```
Comment (terhorstd) [inline src/3.1/entrypoint.sh:37]: If you use the same bash default, then one could also e.g. add modules from the outside. (This is optional and no reason to reject this PR)
```suggestion
    export NEST_SERVER_RESTRICTION_OFF=${NEST_SERVER_RESTRICTION_OFF:-true}
    export NEST_SERVER_MODULES=${NEST_SERVER_MODULES:-nest,numpy}
```

File: src/latest/entrypoint.sh [hunk 11]
```diff
@@ -15,40 +15,32 @@ export LD_LIBRARY_PATH=${MUSIC_PATH}/lib:$LD_LIBRARY_PATH
...
+elif [[ "${MODE}" = 'nest-desktop' ]]; then
+    exec /root/.local/bin/nest-desktop start -h 0.0.0.0 -p 8000
 
-if [[ "$1" = 'nest-desktop' ]]; then
-    cd /opt/data
-    exec /root/.local/bin/nest-desktop start
-fi
+elif [[ "${MODE}" = 'nest-server' ]]; then
+    export NEST_SERVER_RESTRICTION_OFF=true
+    export NEST_SERVER_MODULES=nest,numpy
```
Comment (terhorstd) [inline src/latest/entrypoint.sh:37]: same here, of course.

**Review by terhorstd — APPROVED** [review]
This looks much cleaner! Have a look at one optional idea I commented, but in or not, I think this is good to go. :smiley: 

Thanks for the nice and fast PR! :+1:

**Review by steffengraber — APPROVED** [review]
:+1:  from my side.
I will test everything and then merge.
Thanks @babsey  and @terhorstd  for your great work.

**Pull request merged**